Library/Homebrew/cmd/upgrade.rb

```diff
@@ -171,4 +181,145 @@ def upgrade_formula(f)
       nil
     end
   end
+
+  def upgradable_dependents(kegs, formulae)
+    formulae_to_upgrade = {}
+    formulae_pinned = {}
```

Comment:
Might be worth making these `Set`s instead; `Formulary` does its own caching and it'll ease correct merging.